.github/workflows/test-build.yml

```diff
@@ -0,0 +1,41 @@
+name: ci
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+concurrency:
+  # Cancel any running workflow for the same branch when new commits are pushed.
+  # We group both by ref_name (available when CI is triggered by a push to a branch/tag)
+  # and head_ref (available when CI is triggered by a PR).
+  group: "${{ github.ref_name }}-${{ github.head_ref }}"
+  cancel-in-progress: true
+
+jobs:
+  test-build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go-version: ['1.20', '1.21.x']
```

Comment:
Do we need to run all of this on both Go versions? The Drone pipeline runs everything with the most recent Go version, with the exception of the unit tests which are run on both.

Reply:
Also, any reason to use `1.20` and `1.21.x` (rather than say `1.20.x` and `1.21.x`)?

Reply:
The original drone pipeline runs `make test` in the two versions too. I guess to make sure dskit run fine in those last two go minor versions. I changed the matrix to use `1.20.x` and `1.21.x` for consistency.

Reply:
To be honest, I am not sure what are the differences between `1.20` vs `1.20.x` in this context. I believe both should use the latest patch versions.

Reply:
> The original drone pipeline runs `make test` in the two versions too. I guess to make sure dskit run fine in those last two go minor versions. I changed the matrix to use `1.20.x` and `1.21.x` for consistency.

Yep, but it doesn't run all of the other targets on both versions - only the unit tests are run on both versions. Do we need to run all the targets on both versions? Seems unnecessary.

Reply:
Do you have any suggestions to run the test step with different golang versions?

In Drone, we can define container image in the step level, hence we can have a Go container with a different Golang version running the step, one for 1.20 and another for 1.21. However, in Github action we can only define runner at the job level and there is no thing that is called "go runner". The canonical way to have Golang CI is by using actions/setup-go and then Github action provide a matrix strategy to run the combination of versions across jobs and will run the jobs in parallel. 

Stepping back from talking about test in multiple Golang versions, do you think we need to do that at all? Or should we just one latest go version in the pipeline?

Reply:
>Stepping back from talking about test in multiple Golang versions, do you think we need to do that at all? Or should we just one latest go version in the pipeline?

We have projects that consume dskit that use different versions of Go, so I believe the intent of this is to ensure that we don't start relying on something added in (for example) Go 1.21 when some projects still use 1.20.

>Do you have any suggestions to run the test step with different golang versions?

We'll need to define two separate jobs: one that runs the unit tests with different Go versions with a matrix like you have here, and another job that runs the other targets on just one Go version.